packages/web/src/common/store/cache/sagas.ts

```diff
@@ -105,20 +110,25 @@ export function* retrieve({
   deleteExistingEntry = false,
   onBeforeAddToCache = function* (_metadatas) {},
   onAfterAddToCache = function* (_metadatas) {}
-}: RetrieveArgs) {
+}: RetrieveArgs<T>): Generator<
+  any,
+  { entries: Record<ID | string, T>; uids: Record<ID | string, string> }
+> {
   if (!ids.length) {
     return {
-      entries: [],
-      uids: []
+      entries: {},
+      uids: {}
     }
   }
 
   const uniqueIds = [...new Set(ids)]
   // Create uids for each id and collect a mapping.
-  const uids = makeUids([kind], uniqueIds).reduce<string[]>((map, uid, i) => {
+  const uids = makeUids([kind], uniqueIds).reduce<
+    Record<number | string, string>
+  >((map, uid, i) => {
     map[uniqueIds[i]] = uid
     return map
-  }, [])
+  }, {})
```

Comment:
this was a bugfix

We were using a `string[]` as if it were a `Record<number, string>` because numeric indexes still technically work

Reply:
🐛 🔨